sycl/source/handler.cpp

```diff
@@ -837,11 +860,25 @@ void handler::extractArgsAndReqsFromLambda(
 // Calling methods of kernel_impl requires knowledge of class layout.
 // As this is impossible in header, there's a function that calls necessary
 // method inside the library and returns the result.
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+detail::string handler::getKernelName() {
+  detail::string Str;
+  std::string Fname = MKernel->get_info<info::kernel::function_name>();
+  Str.allocate(Fname.length() + 1);
+  Str.unmarshall(Fname);
+  return Str;
```

Comment:
With proposed changes, we could:
```suggestion
  return detail::string(MKernel->get_info<info::kernel::function_name>());
```